src/server/AuctionHouseServer/AuctionHouse.API/Controllers/UsersController.cs

```diff
@@ -11,5 +13,30 @@ namespace AuctionHouse.API.Controllers
     [ApiController]
     public class UsersController : ControllerBase
     {
+        private readonly IRegisterUserService _registerUser;
+
+        public UsersController(IRegisterUserService registerUser)
+        {
+            _registerUser = registerUser;
+        }
+
+        [HttpPost]
+        public bool Post([FromBody] UserCredentials userCredentials)
+        {
+            UserDto dto = new UserDto(userCredentials);
+
+            if (_registerUser.RegisterUser(dto))
+            {
+                _registerUser.GetUserList().Add(dto);
```

Comment:
Search for "Demeter principle" exposing internals is really bad idea. 

something like this should be enough:

_registerService.RegisterUser(userCredentials);
return Ok(); 

additionaly move it to try/catch block, catch exceptions and return proper http status code

Reply:
Also it's replicated logic from users service